app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -0,0 +1,102 @@
+require 'command_class'
+
+module Authentication
+  module AuthnAzure
+
+    Log = LogMessages::Authentication
+    Err = Errors::Authentication::AuthnAzure
+    # Possible Errors Raised: RoleNotFound, InvalidApplicationIdentity
+
+    ValidateApplicationIdentity = CommandClass.new(
+        dependencies: {
+            role_class:                 Role,
+            resource_class:             Resource,
```

Comment:
Please add rooting (e.g `::Role`). It looks like `Role`, `Resource` & `ApplicationIdentity` are in the same level

Reply:
As explained, if Role was in Authentication > AuthnAzure, it would grab that module so you are correct it is not clear